core/src/main/java/org/elasticsearch/action/update/UpdateResponse.java

```diff
@@ -43,14 +40,28 @@ public UpdateResponse() {
      * Constructor to be used when a update didn't translate in a write.
      * For example: update script with operation set to none
      */
-    public UpdateResponse(ShardId shardId, String type, String id, long version, boolean created) {
-        this(new ShardInfo(0, 0), shardId, type, id, version, created);
+    public UpdateResponse(ShardId shardId, String type, String id, long version, Operation operation) {
+        this(new ShardInfo(0, 0), shardId, type, id, version, operation);
     }
 
-    public UpdateResponse(ShardInfo shardInfo, ShardId shardId, String type, String id, long version, boolean created) {
-        super(shardId, type, id, version);
+    public UpdateResponse(ShardInfo shardInfo, ShardId shardId, String type, String id,
+                          long version, Operation operation) {
+        super(shardId, type, id, version, operation);
         setShardInfo(shardInfo);
-        this.created = created;
+    }
+
+    public static Operation convert(UpdateHelper.Operation op) {
```

Comment:
I wonder if with this change you can remove `UpdateHelper.Operation` entirely and just use `DocWriteResult.Operation`. I'm not sure it'd be clear to use `CREATE` instead of `UPSERT` in all the places though.